File: Easy/Python/Remove-Element.py
1class Solution(object):
2    def removeElement(self, nums, val):
3        k = 0  # point 
4        for i in range(len(nums)): # i is value not index 
5            if nums[i] != val: #curr not val 
6                nums[k] = nums[i]  # shift to k 
7                k += 1 #inc
8        return k  # New length of nums without `val` elements

File: Easy/Java/Remove-Element.java
1class Solution {
2    public int removeElement(int[] nums, int val) {
3        int p = nums.length -1;
4        for (int i = 0; i <= p  ; i++) {
5            if (nums[i] == val) {
6                nums[i] = nums[p];
7                p--;
8                i--;
9            }
10        }
11        return p + 1;
12    }
13}